Providers/Modules/Plugins/Antimalware/plugin/collectdefenderinfo.rb

```diff
@@ -0,0 +1,92 @@
+require "rexml/document"
+require "cgi"
+require 'digest'
+require 'json'
+require 'date'
+require 'time'
+require 'logger'
+require_relative 'antimalwarecommon'
+
+class Defender
+
+	def self.detect()	
+		begin
+			if !File.file?('/usr/bin/mdatp')
+				return false
+			end	
+			return true
+		rescue => e					
+			return false			
+		end
+	end
+
+	def self.getprotectionstatus()		 
+		ret = {}
+		
+		defenderName = "Defender Endpoint Protection for Linux"
+		defenderVersion = "NA"
+		engineVersion = "NA"
+
+		realtimeprotection = "NA"
+		signatureVersion = "NA"
+
+		protectionStatusDetails = ""
+		protectionStatusDetailsString = ""
+
+		error = ""
+
+		($ProtectionStatusRank, $ProtectionStatus) = AntimalwareCommon::UnknownProtectionCode
+		($ThreatStatusRank, $ThreatStatus) = AntimalwareCommon::UnknownThreatCode
+
+		begin
+			json_data = `/usr/bin/mdatp health --output json 2>&1`
+
+			if !$?.success? || json_data.nil? || json_data.empty?				
+				error += "fail to run mdatp health cmd; "
+			end
+
+			defenderhealth = JSON.parse(json_data)
+
+			signatureStatus = defenderhealth["definitionsStatus"]["$type"]
+
+			if signatureStatus != "upToDate"
+				($ProtectionStatusRank, $ProtectionStatus) = AntimalwareCommon::SignaturesOutOfDateProtectionCode
+				protectionStatusDetailsString += "Signatures out of date; "
+			end
+
+			signatureVersion = defenderhealth["definitionsVersion"]
+			defenderVersion = defenderhealth["appVersion"]
+			realtimeprotection = defenderhealth["realTimeProtectionEnabled"]["value"]
+
+			if (realtimeprotection == "NA" || !realtimeprotection)
+				($ProtectionStatusRank, $ProtectionStatus) = AntimalwareCommon::NoRealTimeProtectionProtectionCode
+				protectionStatusDetailsString += "Real-time protection disabled; "
+			end
+
+			if protectionStatusDetailsString.empty? 
+				($ProtectionStatusRank, $ProtectionStatus) = AntimalwareCommon::RealTimeProtectionCode
+				protectionStatusDetailsString += "MDATP is running healthy."
+			end
+			protectionStatusDetails = protectionStatusDetailsString
```

Comment:
This needs to be in else part. otherwise you are parsing json_data in error scenario too

Reply:
I've moved all data parsing/acquisition underneath an `else` in the event the command ran was successful.
If the json fails to parse, it should be caught in the `rescue` if I understand correctly